domain/port/state/watcher.go

```diff
@@ -62,31 +66,30 @@ WHERE unit_endpoint.uuid IN ($endpoints[:])
 	return transform.Slice(machineNames, func(m machineName) coremachine.Name { return m.Name }), nil
 }
 
-// FilterEndpointsForApplication returns the subset of provided endpoint uuids
-// that are associated with the provided application.
-func (st *State) FilterEndpointsForApplication(ctx context.Context, app coreapplication.ID, eps []string) (set.Strings, error) {
+// GetApplicationNamesForEndpoints returns a slice of application names that host
```

Comment:
Even though it makes it longer still, I'm inclined to name this `GetApplicationNamesForUnitEndpoints`, just because I had to read through to get what it meant. My inclination was to think of this as a charm endpoint, rather than the unique entities we designate as being per unit.

Reply:
Agree